app/Core/Router.php

```diff
@@ -27,6 +33,54 @@ private function __construct() {
      * @return Router
      */
     public static function getInstance() : Router {
-        return self::$routerInstance ?? new Router();
+        return $this->$routerInstance ?? new Router();
+    }
+
+    public function parseUrl($url = ''){
```

Comment:
This function has no business here. Remove it and check out my last comment for better explanation of why.
And the same goes for the `getController()`, `getAction()` and `getParams()`